src/ModuleLoader.ts

```diff
@@ -672,13 +671,11 @@ export class ModuleLoader {
 			} as ResolvedId;
 		}
 		if (resolution == null) {
-			return (module.resolvedIds[specifier] =
-				module.resolvedIds[specifier] ||
-				this.handleResolveId(
-					await this.resolveId(specifier, module.id, EMPTY_OBJECT, false),
-					specifier,
-					module.id
-				));
+			return (module.resolvedIds[specifier] ??= this.handleResolveId(
```

Comment:
Are we using `??` in the codebase now?

Reply:
yeah, `nullish coalescing` is used throughout the codebase, `logical nullish assignment` a little bit.

down-leveling only adds a tiny amount of additional code which results in almost the same amount without the change. if v3 were about to support node.js v14+ `nullish coalescing` would also run native as well. short circuiting should also make it faster.

the correct syntax mapping would have been `||=` rather, but I thought since `ResolveId` is an object, `??=` would fit better to show the intend of only assigning when `null` or `undefined`.